wdl2cwl/main_oop.py

```diff
@@ -443,29 +439,33 @@ def get_cpu_requirement(self, cpu_runtime: WDL.Expr.Base) -> str:
         return f"$({cpu_str})"
 
     def get_cwl_docker_requirements(
-        self, wdl_docker: Union[WDL.Expr.Get, WDL.Expr.String]
+        self, wdl_docker: WDL.Expr.Base
     ) -> cwl.ProcessRequirement:
         """Translate WDL Runtime Docker requirements to CWL Docker Requirement."""
+        if not isinstance(wdl_docker, (WDL.Expr.Get, WDL.Expr.String)):
+            raise ValueError(
+                f"Unsupported docker runtime type: {type(wdl_docker)}: {wdl_docker}"
+            )
```

Comment:
This was being done before calling the function. But since both check + function are a pair, and must be executed in order (i.e. you first check the instance, and then call the function) these can be merged and have the check inside the function, avoiding possible bugs where we forget to call the check before calling function...